storm-core/src/clj/backtype/storm/daemon/common.clj

```diff
@@ -273,6 +293,26 @@
      (metrics-consumer-register-ids storm-conf)
      (get storm-conf TOPOLOGY-METRICS-CONSUMER-REGISTER))))
 
+;; return the fields that event logger bolt expects
+(defn eventlogger-bolt-fields []
+  [(EventLoggerBolt/FIELD_COMPONENT_ID)  (EventLoggerBolt/FIELD_TS) (EventLoggerBolt/FIELD_VALUES)]
+  )
+
+(defn add-eventlogger! [storm-conf ^StormTopology ret]
+  (let [num-executors (if (nil? (storm-conf TOPOLOGY-EVENTLOGGER-EXECUTORS)) (storm-conf TOPOLOGY-WORKERS) (storm-conf TOPOLOGY-EVENTLOGGER-EXECUTORS))
```

Comment:
may be i am missing this, but if TOPOLOGY-EVENTLOGGER-EXECUTORS is set to 0, shouldn't you just return? I do not see the logic that just does not add event logger bolt when TOPOLOGY-EVENTLOGGER-EXECUTORS = 0.  Or are we assumin ghtat as num-executors will be set to 0, no thread will execute this task?